Bottom: ec436eee5a5a2ea5c8704cb64c01aba0bca8e9f5
Top:    3e5b4cd6ebdacb08a2ba5ede41e07dd127a33d90
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-12 17:50:36 +0100

Increase initial heap size from 32 to 256

In complex boards with lots of polygons, we were realloc'ing up
to this kind of number anyway, in x2 size steps. Save all the
memcpy'ing by making the heap larger to start with.

My test board still has some requirement for heaps >256, <512,
but the frequency is low.



---

diff --git a/src/heap.c b/src/heap.c
index 39dcbe4..4959b58 100644
--- a/src/heap.c
+++ b/src/heap.c
@@ -165,7 +165,7 @@ heap_insert (heap_t * heap, cost_t cost, void *data)
     {
       heap->max *= 2;
       if (heap->max == 0)
-	heap->max = 32;		/* default initial heap size */
+	heap->max = 256;		/* default initial heap size */
       heap->element =
 	realloc (heap->element, heap->max * sizeof (*heap->element));
     }
